### PR TITLE
Update best practices from upstream Chromium docs

### PR DESCRIPTION
**Pull request by netzenbot**
## Summary

Synced best practices with upstream documentation (Chromium C++ style guide, ChromiumOS testing guide).

Scope was narrowed after review to only the uncontested, verifiable items:

- **Fixed outdated platform-check examples (CS-033, CS-034):** replaced deprecated `#if defined(OS_ANDROID)` with `#if BUILDFLAG(IS_ANDROID)`, matching the upstream style guide and our own `build-system.md`. Added explicit guidance to use `build/build_config.h` macros over raw compiler macros.
  - Source: [Chromium C++ style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md) ("To `#ifdef` code for specific platforms, use the macros defined in `build/build_config.h` … not other macros set by specific compilers or build environments (e.g. `WIN32`)"), enforced upstream by `CheckForDeprecatedOSMacros` in `PRESUBMIT.py`.
- **Updated TI-039:** anchored test-coverage guidance to the 80-90% target from the ChromiumOS testing guide.
  - Source: [ChromiumOS testing guide — "How much test coverage is enough?"](https://www.chromium.org/chromium-os/developer-library/guides/testing/cpp-writing-tests/) ("aim for at least 80%, preferably 90% test coverage. Trying for more than 90% test coverage is often counterproductive"). Note this is the ChromiumOS guide, not a Chromium-wide policy — the rule text presents it as a guideline, not a gate.

### Removed from this PR

- **CS-071 (`DumpWithoutCrashing()`)** — dropped per review. Not a Brave best practice; we rely on `DUMP_WILL_BE_NOTREACHED` instead, and the "rate-limited, safe in hot paths" claim was not verified against Backtrace.

The remaining upstream rules (smart pointers, base containers, componentization/layering, and the bulk of the testing guide) were already fully covered by existing best practices.

Docs-only change — no code or test impact.

### CODE REVIEW

File: docs/best-practices/coding-standards.md [hunk 1]
```diff
@@ -1320,3 +1325,43 @@ This is a mojom-specific application of [CS-014](#CS-014). The
 target produces it.
 
 ---
+
+<a id="CS-071"></a>
+
+## ✅ Use `DumpWithoutCrashing()` to Report Unexpected States Without Crashing
```
Comment (bridiver) [inline docs/best-practices/coding-standards.md]: https://bravesoftware.slack.com/archives/C7VLGSR55/p1761195306460749?thread_ts=1760812046.052219&cid=C7VLGSR55

Comment (bridiver) [reply docs/best-practices/coding-standards.md]: but this comes from upstream docs? Would be nice to have the relevant link
cc @goodov

File: docs/best-practices/coding-standards.md [hunk 2]
```diff
@@ -1320,3 +1325,43 @@ This is a mojom-specific application of [CS-014](#CS-014). The
...
+
+// ✅ CORRECT - degrade gracefully AND report for investigation
+std::optional<mojom::AdType> ToMojomAdType(const std::string& type) {
+  // ...
+  base::debug::DumpWithoutCrashing();  // rate-limited; safe in release
+  return std::nullopt;
+}
+```
+
+`DumpWithoutCrashing()` is rate-limited by the crash system, so it is safe in
```
Comment (bridiver) [inline docs/best-practices/coding-standards.md]: I don't know if this is true for backtrace or not

**Comment by bbondy** [comment]
@netzenbot let's just implement the non-questionable things for this PR and remove the rest. Update the original description and ping me when completed.

File: docs/best-practices/coding-standards.md (hunk 1 above)
Comment (goodov) [reply docs/best-practices/coding-standards.md]: that's not a best practice for us. We should rely only on `DUMP_WILL_BE_NOTREACHED`.

Comment (netzenbot) [reply docs/best-practices/coding-standards.md]: Agreed — removed CS-071 from this PR entirely. `DUMP_WILL_BE_NOTREACHED` remains the guidance, so there is nothing new to document here.

File: docs/best-practices/coding-standards.md (hunk 2 above)
Comment (netzenbot) [reply docs/best-practices/coding-standards.md]: Fair point — I could not verify the rate-limiting behaviour for Backtrace, so I dropped the whole CS-071 section rather than ship an unverified claim.

File: docs/best-practices/testing-isolation.md [hunk 3]
```diff
@@ -764,6 +764,12 @@ ensure that new logic, branches, and edge cases have corresponding tests. This
 applies to all PRs, not just bug fixes. Don't be overly strict — use judgment —
 but flag obvious gaps where meaningful test coverage is missing.
 
+As an anchor, the
```
Comment (netzenbot) [inline docs/best-practices/testing-isolation.md:767]: nit: the linked page (`chromium-os/developer-library/guides/testing/cpp-writing-tests/`) is the ChromiumOS testing guide, not Chromium's. The 80%/90% wording matches it verbatim, so the numbers are fine — just relabel the link as "ChromiumOS testing guide" so readers don't read it as Chromium-wide policy. ([✅ Delete Dead Documentation](https://github.com/brave/brave-core/tree/master/docs/best-practices/documentation.md#DOC-002))

Comment (netzenbot) [reply docs/best-practices/testing-isolation.md:767]: Good catch — fixed. The link is the ChromiumOS developer library page, so it is now labelled **ChromiumOS testing guide**. The 80% / preferably 90% numbers are quoted verbatim from that page and are unchanged. Also updated the PR description so it does not present this as Chromium-wide policy.

**Comment by bbondy** [comment]
I don't want to chase this down. closing.
